docs/reference/cpp/index.md

```diff
@@ -0,0 +1,254 @@
+# Getting Started
+
+UBC Sailbot's Network Systems team uses C++ for its software. If you know already know C, then you already know the
+bare minimum to write C++. This is a good starting point, but the additional features C++ provides allow for safer
+programming practices.
+
+## For C/C++ Beginners
+
+If you just need to know how C++ is different from C, then see the [Differences Between C and C++](#differences-between-c-and-c).
+You should also look at it if you go through and finish this section.
+
+If you are new to C and C++, then this the best place to start. The tutorials provided in this section will help you
+learn the fundamentals of the language. Do not feel pressured to do all the tutorials! Just get comfortable with the
+syntax and the mechanisms of the language.
+
+!!! note
+    The hardest part about this will likely be pointers and dynamic memory, so pay close attention to tutorials
+    concerning them! Additionally, dynamic memory requires the usage of pointers, but pointers do not require dynamic
+    memory!
+
+!!! tip
+    Dynamic memory is much more prone to error than statically allocated memory, so try to use static allocation
+    whenever possible
+
+| Resource               | Description                                                                                 |
+| :--------------------- | :------------------------------------------------------------------------------------------ |
+| [w3schools Tutorial](https://www.w3schools.com/cpp/default.asp) | A structured tutorial that goes through basic concepts in C++. It's good to do up to the section on Classes. |
+| [YouTube Tutorial](https://youtu.be/vLnPwxZdW4Y) | If you prefer video tutorial, then this is a comprehensive 4 hour video covering similar concepts to the one above. It is 4 hours long though. |
+| [Dynamic Memory Overview](https://www.tutorialspoint.com/cplusplus/cpp_dynamic_memory.htm) | A page going over how dynamic memory works in C++. |
+
+Feel free to add other resources other than the ones listed above if you find any that you like!
+
+## Differences Between C and C++
+
```

Comment:
This can belong on its own page in another markdown file. Then, add it to the subtree under C++.